internal/attestation/azure/snp/validator.go

```diff
@@ -52,235 +55,157 @@ func NewValidator(cfg *config.AzureSEVSNP, log attestation.Logger) *Validator {
-	if err != nil {
-		return nil, fmt.Errorf("validating VCEK: %w", err)
-	}
-
-	if err := v.validateSNPReport(ctx, vcek, report, instanceInfo.MAAToken, extraData); err != nil {
-		return nil, fmt.Errorf("validating SNP report: %w", err)
-	}
-
+	// Verify the attestation report's certificates.
+	if err := verify.SnpAttestation(att, &verify.Options{}); err != nil {
+		return nil, fmt.Errorf("verifying SNP attestation: %w", err)
+	}
+
+	// If the enforcement policy is set to MAAFallback or WarnOnly, the check of the IDKey hashes should not be enforced.
+	requireIDBlock := !(v.config.FirmwareSignerConfig.EnforcementPolicy == idkeydigest.MAAFallback ||
+		v.config.FirmwareSignerConfig.EnforcementPolicy == idkeydigest.WarnOnly)
+
+	// Checks if the attestation report matches the given constraints.
+	// Some constraints are implicitly checked by validate.SnpAttestation:
+	// - the report is not expired
+	if err := validate.SnpAttestation(att, &validate.Options{
+		GuestPolicy: abi.SnpPolicy{
+			Debug: false, // Debug means the VM can be decrypted by the host for debugging purposes and thus is not allowed.
+			SMT:   true,  // Allow Simultaneous Multi-Threading (SMT). Normally, we would want to disable SMT
+			// but Azure does not allow to disable it.
+		},
+		// This checks that the reported TCB version is equal or greater than the minimum specified in the config.
+		MinimumTCB: kds.TCBParts{
+			BlSpl:    v.config.BootloaderVersion.Value, // Bootloader
+			TeeSpl:   v.config.TEEVersion.Value,        // TEE (Secure OS)
+			SnpSpl:   v.config.SNPVersion.Value,        // SNP
+			UcodeSpl: v.config.MicrocodeVersion.Value,  // Microcode
+		},
+		// This checks that the reported LaunchTCB version is equal or greater than the minimum specified in the config.
+		MinimumLaunchTCB: kds.TCBParts{
+			BlSpl:    v.config.BootloaderVersion.Value, // Bootloader
+			TeeSpl:   v.config.TEEVersion.Value,        // TEE (Secure OS)
+			SnpSpl:   v.config.SNPVersion.Value,        // SNP
+			UcodeSpl: v.config.MicrocodeVersion.Value,  // Microcode
+		},
+		// Check that CurrentTCB >= CommittedTCB.
+		PermitProvisionalFirmware: true,
+		// Check if the IDKey hashes in the report are in the list of accepted hashes.
+		TrustedIDKeyHashes: v.config.FirmwareSignerConfig.AcceptedKeyDigests,
+		// The IDKey hashes should not be checked if the enforcement policy is set to MAAFallback or WarnOnly to prevent
+		// an error from being returned because of the TrustedIDKeyHashes validation. In this case, we should perform a
+		// custom check of the MAA-specific values later. Right now, this is a double check, since a custom MAA check
+		// is performed either way.
+		RequireIDBlock: requireIDBlock,
+	}); err != nil {
+		return nil, fmt.Errorf("validating SNP attestation: %w", err)
+	}
+	if requireIDBlock {
+		// Custom WarnOnly / MAAFallback check of the IDKeyDigests.
+		v.checkIDKeyDigest(ctx, att, instanceInfo.MAAToken, extraData)
```

Comment:
```suggestion
		// Custom WarnOnly / MAAFallback check of the IDKeyDigests.
		if err := v.checkIDKeyDigest(ctx, att, instanceInfo.MAAToken, extraData); err != nil {
			return err
		}
```
Return value not checked